crates/solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -289,16 +291,50 @@ impl<'a> FlashbotsSolutionSubmitter<'a> {
                 gas_estimate,
             );
 
-            match self
-                .flashbots_api
-                .submit_transaction(&raw_signed_transaction)
-                .await
-            {
-                Ok(bundle_id) => {
-                    transactions.push(hash);
-                    previous_tx = Some((gas_price, bundle_id));
+            // submit transaction
+
+            let mut retry = std::cmp::max(SUBMIT_RETRY_ATTEMPTS, 1usize);
+            loop {
+                // decrease number of attempts
+                if retry == 0 {
+                    break;
+                } else {
+                    retry -= 1;
                 }
-                Err(err) => tracing::error!("flashbots submission failed: {:?}", err),
+
+                // submit tx and get handle - bundle_id
+                let bundle_id = match self
+                    .flashbots_api
+                    .submit_transaction(&raw_signed_transaction)
+                    .await
+                {
+                    Ok(bundle_id) => {
+                        transactions.push(hash);
+                        previous_tx = Some((gas_price, bundle_id.clone()));
+                        bundle_id
+                    }
+                    Err(err) => {
+                        tracing::error!("flashbots submission failed: {:?}", err);
+                        break;
+                    }
+                };
+
+                // handle retry based on the current status of the submitted tx
+                match self.flashbots_api.is_transaction_failed(&bundle_id).await {
```

Comment:
What cases does this cover? If the transaction is no longer simulating properly we don't want to retry it do we?

Reply:
This is a bad attempt to solve the cancellation issue we are having with flashbots.
I've created a new function `cancel_and_wait` and the code should be a bit more clear and readable now, and also a bit more convenient to change to our "general submitting strategy" that we are aiming to build in, guess, future refactor PR :smile: